src/clients/SpokePoolClient.ts

```diff
@@ -573,9 +577,13 @@ export class SpokePoolClient extends BaseAbstractClient {
     });
 
     const timerStart = Date.now();
-    const [numberOfDeposits, currentTime, ...events] = await Promise.all([
+    // TODO: Once the SpokePools are updated and have the `fillDeadlineBuffer()` method, load it dynamically here.
+    const fillDeadlineBuffer = 2 * 60 * 60; // 2 hours. We're assuming this is hardcoded to the same value that the
```

Comment:
What happens if the fill deadline buffer changes?

There may be nothing we can do to guard against that other than make sure we add a cushion here and don't change it dramatically at any one time.

Reply:
I don't know how to handle it since the `fillDeadlineBuffer` is a constant/immutable in the contract so we don't have an easy to track its updates. We could try to track Upgraded events but its not super easy. I also think the value here doesn't need to align with the contracts exactly, as long as its no smaller.

Reply:
I'll add a comment here to that^

Reply:
thoughts on this treatment? 2a3963097a3cbd56ee1d0e9349e33a5456fde5fc

Reply:
The latest implementation now removes `fillDeadlineBuffer` from being loaded by the `SpokePoolClient`. Instead, we load it at the `dataworker` time by querying `spokePool.fillDeadlineBuffer({ blockTag: start|end })` and taking the maximum of the value at the start and end of a block range. 

(We assume this value doesn't change in the middle of a bundle, which is a fair assumption 99% of the time and even if its not correct, probably still protects us if the admin updating `fillDeadlineBuffer` is reasonable.

https://github.com/across-protocol/relayer-v2/blob/017f51a833d4f5504522b6e546903a0ad912804d/src/dataworker/DataworkerUtils.ts#L100

Reply:
Yeah, I think that's the best we can do.

Reply:
Thanks for adding that comment -- I agree. I also forgot that it was immutable, so probably even less likely to be changed.